core/src/main/java/org/elasticsearch/cluster/routing/AllocationId.java

```diff
@@ -148,12 +156,52 @@ public String toString() {
 
     @Override
     public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
-        builder.startObject("allocation_id");
-        builder.field("id", id);
+        builder.startObject(ALLOCATION_ID_OBJECT_KEY);
+        builder.field(ID_KEY, id);
         if (relocationId != null) {
-            builder.field("relocation_id", relocationId);
+            builder.field(RELOCATION_ID_KEY, relocationId);
         }
         builder.endObject();
         return builder;
     }
+
+    public static AllocationId fromXContent(XContentParser parser) throws IOException {
```

Comment:
not sure how DocumentParser helps here. Extra class Fields with XContentBuilderString does not help here as we match exact string names in fromXContent method (Fields is only used in cases where there is only toXContent). I implemented it in exactly the same way as it's done for ShardStateMetaData ...